File: app/src/main/java/com/example/trackrate/util/PasswordPolicy.kt
package com.example.trackrate.util

object PasswordPolicy {

    const val ERROR_MESSAGE =
        "La contraseña debe tener al menos 12 caracteres, una letra, un número y un carácter especial."

    fun validationError(password: String): String? {
        val isValid =
            password.length >= 12 &&
                    password.any { it.isLetter() } &&
                    password.any { it.isDigit() } &&
                    password.any {
                        !it.isLetterOrDigit() && !it.isWhitespace()
                    }

        return if (isValid) null else ERROR_MESSAGE
    }
}

File: backend/app/core/password_policy.py
import re

MIN_PASSWORD_LENGTH = 12


def validate_password(password: str) -> None:
    if len(password) < MIN_PASSWORD_LENGTH:
        raise ValueError(
            f"La contraseña debe tener al menos {MIN_PASSWORD_LENGTH} caracteres"
        )
    if not re.search(r"[A-Za-z]", password):
        raise ValueError("La contraseña debe incluir al menos una letra")
    if not re.search(r"\d", password):
        raise ValueError("La contraseña debe incluir al menos un número")
    if not re.search(r"[^A-Za-z0-9]", password):
        raise ValueError("La contraseña debe incluir al menos un carácter especial")
